File: mobile/src/domain/core/status.ts
import { User } from "./user"

export enum FFStatus {
  ON = "ON",
  OFF = "OFF",
}

export type FFValue = FFStatus | string

export enum LoginStatusCode {
  LOCKED = "LOCKED",
  UNLOCKED = "UNLOCKED",
}

export enum BackendLogLevel {
  NONE = "NONE",
  DEBUG = "DEBUG",
  INFO = "INFO",
  WARNING = "WARNING",
  ERROR = "ERROR",
  CRITICAL = "CRITICAL",
}

export interface BackendOptions {
  dataDir?: string | null
  port?: number | null
  logLevel?: BackendLogLevel | null
  logDir?: string | null
  logFileLevel?: BackendLogLevel | null
  thirdPartyLogLevel?: BackendLogLevel | null
}

export interface BackendDetails {
  version: string
  options: BackendOptions
}

export interface GlobalStatus {
  status: LoginStatusCode
  server: BackendDetails
  features: Record<string, FFStatus | string>
  user?: User | null
  lastLogged?: string | null
}

File: finanze/domain/status.py
from dataclasses import dataclass
from enum import Enum
from typing import Optional, TypeAlias

from domain.user import User


class FFStatus(str, Enum):
    ON = "ON"
    OFF = "OFF"


FFValue = FFStatus | str

FeatureFlags: TypeAlias = dict[str, FFValue]


class LoginStatusCode(str, Enum):
    LOCKED = "LOCKED"
    UNLOCKED = "UNLOCKED"


class BackendLogLevel(str, Enum):
    NONE = "NONE"
    DEBUG = "DEBUG"
    INFO = "INFO"
    WARNING = "WARNING"
    ERROR = "ERROR"
    CRITICAL = "CRITICAL"


@dataclass
class BackendOptions:
    data_dir: Optional[str] = None
    port: Optional[int] = None
    log_level: Optional[BackendLogLevel] = None
    log_dir: Optional[str] = None
    log_file_level: Optional[BackendLogLevel] = None
    third_party_log_level: Optional[BackendLogLevel] = None


@dataclass
class BackendDetails:
    version: str
    options: BackendOptions


@dataclass
class GlobalStatus:
    status: LoginStatusCode
    server: BackendDetails
    features: FeatureFlags
    user: Optional[User] = None
    last_logged: Optional[str] = None
